torch/distributed/_tensor/dispatch.py

```diff
@@ -302,6 +302,7 @@ def unwrap_to_op_info(
                 args_schema.append(arg._spec)
                 local_args.append(arg._local_tensor)
                 if mesh is not None:
+                    print(f"{mesh=}, {arg.device_mesh=}")
```

Comment:
I tried repro'ing the test_dtensor_compile failure locally. It's actually coming from here - the test passes when I remove the print statement.

I mostly could tell by looking at the stacktrace from the FakeTensor erroring, and seeing that it's coming from:

(1) this code is printing `arg.device_mesh`

(2) `DeviceMesh` is a tensor, and printing it calls `tensor.toList()`

(3) printing a tensor is not a very trace-friendly operation... which is why you get a kind-of-obscure error